test/unit/shim/shim.test.js

```diff
@@ -541,6 +541,31 @@ tap.test('Shim', function (t) {
       t.equal(shim.unwrap(wrappable.bar), original)
       t.end()
     })
+
+    t.test('should wrap child instance properly', function (t) {
+      class ParentTest {
+        constructor() {
+          this.parent = true
```

Comment:
Just based on the name of this property, we should probably set it to `false` in the child class and then check that. Or, remove this property since it's never checked anywhere.